pages/tokens.py

```diff
@@ -122,24 +170,24 @@ def generate_tokens(n_clicks, program, token_length, token_count, out_format, ch
     Input('token-export', 'n_clicks'),
 )
 def export_tokens(n_clicks):
-    def zip_directory(bytes_io):
+    def zip_directory():
+        bytes_io = io.BytesIO()
         with zipfile.ZipFile(bytes_io, mode="w") as zf:
-            len_dir_path = len(QRCODE_PATH)
-            for root, _, files in os.walk(QRCODE_PATH):
-                for img in files:
-                    file_path = os.path.join(root, img)
-                    zf.write(file_path, file_path[len_dir_path:])
-
+            for token in edb.get_token_db().find({}, {"token": 1}):
+                img_bytes = saveAsQRCode(token['token'])
+                zf.writestr(f"{token['token']}.png", img_bytes.getvalue())
+        bytes_io.seek(0)
+        return bytes_io.read()
     if n_clicks > 0:
-        return dcc.send_bytes(zip_directory, "tokens.zip")
+        return dcc.send_bytes(zip_directory(), "tokens.zip")
```

Comment:
It isn't clear to me why this was changed from `zip_directory` to `zip_directory()`. Can you clarify?

Reply:
Here is my interpretation, let me know if this is incorrect-

If you pass a plain function to `dcc.send_bytes`, it calls that function and provides a `bytes_io` as the argument to your function.
But instead of accepting that argument, we are now creating our own `bytes_io` and returning that.

Reply:
When we pass a function to ```dcc.send_bytes```, Dash calls that function and expects it to return bytes directly. In this case, ```zip_directory``` is a function that returns bytes, so using ```zip_directory()``` correctly calls the function and provides the bytes directly to dcc.send_bytes. This ensures that the bytes of the zipped directory are correctly sent for download as a zip file.